system/lib/html5/offscreencanvas.c

```diff
@@ -3,23 +3,27 @@
 #include "emscripten_internal.h"
 
 typedef struct set_cavas_size_t {
-  const char* target;
+  char* target;
   int width;
   int height;
 } set_cavas_size_t;
 
 static void do_set_size(void* arg) {
   set_cavas_size_t* args = (set_cavas_size_t*)arg;
   _emscripten_set_offscreencanvas_size(args->target, args->width, args->height);
+  free(args->target);
   free(arg);
 }
 
+// Warning: this function takes ownership of the "target" parameter
+// - DO NOT call this function with a string literal
+// - DO NOT free the memory after calling this function
```

Comment:
I'm not sure we need to this much emphasis here.   How about just `This function takes ownership of the "target" string`.